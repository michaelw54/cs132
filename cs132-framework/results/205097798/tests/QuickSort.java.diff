1,4c1,22
< ParseException: Encountered " <IDENTIFIER> "java "" at line 216, column 1.
< Was expecting:
<     "return" ...
<     
---
> 20
> 7
> 12
> 18
> 2
> 11
> 6
> 9
> 19
> 5
> 9999
> 2
> 5
> 6
> 7
> 9
> 11
> 12
> 18
> 19
> 20
> 0
