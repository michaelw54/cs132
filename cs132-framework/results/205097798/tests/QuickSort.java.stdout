func Main()
    w4 = 12
    w2 = alloc(w4)
    w4 = 16
    w3 = alloc(w4)
    [w2 + 0] = w3
    w4 = @QS_Print
    [w3 + 0] = w4
    w4 = @QS_Init
    [w3 + 4] = w4
    w4 = @QS_Start
    [w3 + 8] = w4
    w4 = @QS_Sort
    [w3 + 12] = w4
    w5 = 10
    w1 = @QS_Start
    w0 = call w1(w2 w5)
    print(w0)
    ret_ = 0
    return ret_
    
func QS_Start(this sz)
    aux01 = 0
    w9 = this
    w10 = sz
    w8 = @QS_Init
    w7 = call w8(w9 w10)
    aux01 = w7
    w15 = this
    w14 = @QS_Print
    w13 = call w14(w15)
    aux01 = w13
    w18 = 9999
    print(w18)
    w20 = [this + 8]
    w22 = 1
    w19 = w20 - w22
    aux01 = w19
    w27 = this
    w28 = 0
    w29 = aux01
    w26 = @QS_Sort
    w25 = call w26(w27 w28 w29)
    aux01 = w25
    w34 = this
    w33 = @QS_Print
    w32 = call w33(w34)
    aux01 = w32
    w37 = 0
    return w37
    
func QS_Sort(this left right)
    v = 0
    i = 0
    j = 0
    nt = 0
    t = 0
    cont01 = 0
    cont02 = 0
    aux03 = 0
    w38 = 0
    t = w38
    w41 = left
    w43 = right
    w40 = w41 < w43
    if0 w40 goto else_40
    w49 = [this + 4]
    w50 = right
    w46 = 1
    w47 = w50 + w46
    w46 = [w49 + 0]
    w48 = w47 < w46
    if0 w48 goto outOfBounds_45
    w46 = 4
    w47 = 1
    w48 = w50 + w47
    w47 = w48 * w46
    w46 = w49 + w47
    w45 = [w46 + 0]
    goto pass_45
  outOfBounds_45:
    error("array index out of bounds")
  pass_45:
    v = w45
    w54 = left
    w56 = 1
    w53 = w54 - w56
    i = w53
    w59 = right
    j = w59
    w61 = 1
    cont01 = w61
  loop_63:
    w63 = cont01
    if0 w63 goto endLoop_63
    w64 = 1
    cont02 = w64
  loop_66:
    w66 = cont02
    if0 w66 goto endLoop_66
    w68 = i
    w70 = 1
    w67 = w68 + w70
    i = w67
    w77 = [this + 4]
    w78 = i
    w74 = 1
    w75 = w78 + w74
    w74 = [w77 + 0]
    w76 = w75 < w74
    if0 w76 goto outOfBounds_73
    w74 = 4
    w75 = 1
    w76 = w78 + w75
    w75 = w76 * w74
    w74 = w77 + w75
    w73 = [w74 + 0]
    goto pass_73
  outOfBounds_73:
    error("array index out of bounds")
  pass_73:
    aux03 = w73
    w84 = aux03
    w86 = v
    w83 = w84 < w86
    w82 = 1
    w81 = w83 < w82
    if0 w81 goto else_81
    w88 = 0
    cont02 = w88
    goto end_81
  else_81:
    w90 = 1
    cont02 = w90
  end_81:
    goto loop_66
  endLoop_66:
    w92 = 1
    cont02 = w92
  loop_94:
    w94 = cont02
    if0 w94 goto endLoop_94
    w96 = j
    w98 = 1
    w95 = w96 - w98
    j = w95
    w105 = [this + 4]
    w106 = j
    w102 = 1
    w103 = w106 + w102
    w102 = [w105 + 0]
    w104 = w103 < w102
    if0 w104 goto outOfBounds_101
    w102 = 4
    w103 = 1
    w104 = w106 + w103
    w103 = w104 * w102
    w102 = w105 + w103
    w101 = [w102 + 0]
    goto pass_101
  outOfBounds_101:
    error("array index out of bounds")
  pass_101:
    aux03 = w101
    w112 = v
    w114 = aux03
    w111 = w112 < w114
    w110 = 1
    w109 = w111 < w110
    if0 w109 goto else_109
    w116 = 0
    cont02 = w116
    goto end_109
  else_109:
    w118 = 1
    cont02 = w118
  end_109:
    goto loop_94
  endLoop_94:
    w124 = [this + 4]
    w125 = i
    w121 = 1
    w122 = w125 + w121
    w121 = [w124 + 0]
    w123 = w122 < w121
    if0 w123 goto outOfBounds_120
    w121 = 4
    w122 = 1
    w123 = w125 + w122
    w122 = w123 * w121
    w121 = w124 + w122
    w120 = [w121 + 0]
    goto pass_120
  outOfBounds_120:
    error("array index out of bounds")
  pass_120:
    t = w120
    w132 = i
    w137 = [this + 4]
    w138 = j
    w134 = 1
    w135 = w138 + w134
    w134 = [w137 + 0]
    w136 = w135 < w134
    if0 w136 goto outOfBounds_133
    w134 = 4
    w135 = 1
    w136 = w138 + w135
    w135 = w136 * w134
    w134 = w137 + w135
    w133 = [w134 + 0]
    goto pass_133
  outOfBounds_133:
    error("array index out of bounds")
  pass_133:
java.lang.NullPointerException
